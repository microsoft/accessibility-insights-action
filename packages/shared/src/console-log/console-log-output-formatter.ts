// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

import { injectable } from 'inversify';
import { toolName } from '../content/strings';
import { OutputFormatter } from '../output/output-formatter';
@injectable()
export class ConsoleLogOutputFormatter implements OutputFormatter {
    public escaped(text: string): string {
        return text;
    }
    public snippet(text: string): string {
        return `\`${text}\``;
    }
    public link(href: string, text: string): string {
        return `${text}: ${href}`;
    }
    public image(altText: string, src: string): string {
        return `![${altText}](${src})`;
    }
    public listItem(text: string): string {
        return `* ${text}`;
    }
    // eslint-disable-next-line @typescript-eslint/no-unused-vars
    public heading(text: string, headingLevel: number): string {
        return text;
    }
    public bold(text: string): string {
        return text;
    }
    public productTitle(): string {
        return `${toolName}`;
    }
    public footerSeparator(): string {
        return `----------------------------------------`;
    }
    public sectionSeparator(): string {
        return '\n';
    }
}
