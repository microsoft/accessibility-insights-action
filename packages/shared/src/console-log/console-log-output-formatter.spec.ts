// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import 'reflect-metadata';
import { ConsoleLogOutputFormatter } from './console-log-output-formatter';

describe('ConsoleLogOutputFormatter', () => {
    const testSubject = new ConsoleLogOutputFormatter();
    it('escaped', () => {
        expect(testSubject.escaped('<img>')).toEqual('\\<img>');
    });

    it('snippet', () => {
        expect(testSubject.snippet('code')).toMatchSnapshot();
    });

    it('link', () => {
        expect(testSubject.link('href', 'text')).toMatchSnapshot();
    });

    it('image', () => {
        expect(testSubject.image('alt', 'src')).toMatchSnapshot();
    });

    it('listItem', () => {
        expect(testSubject.listItem('li')).toMatchSnapshot();
    });

    it('heading', () => {
        expect(testSubject.heading('heading', 2)).toMatchSnapshot();
    });

    it('bold', () => {
        expect(testSubject.bold('text')).toMatchSnapshot();
    });

    it('productTitle', () => {
        expect(testSubject.productTitle()).toMatchSnapshot();
    });

    it('footerSeparator', () => {
        expect(testSubject.footerSeparator()).toMatchSnapshot();
    });

    it('sectionSeparator', () => {
        expect(testSubject.sectionSeparator()).toMatchSnapshot();
    });
});
