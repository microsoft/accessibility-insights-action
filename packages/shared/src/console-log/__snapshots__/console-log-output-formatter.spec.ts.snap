// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsoleLogOutputFormatter bold 1`] = `"text"`;

exports[`ConsoleLogOutputFormatter footerSeparator 1`] = `"----------------------------------------"`;

exports[`ConsoleLogOutputFormatter heading 1`] = `"heading"`;

exports[`ConsoleLogOutputFormatter image 1`] = `"Image: alt (src)"`;

exports[`ConsoleLogOutputFormatter link 1`] = `"text: href"`;

exports[`ConsoleLogOutputFormatter listItem 1`] = `"* li"`;

exports[`ConsoleLogOutputFormatter productTitle 1`] = `"Accessibility Insights Action"`;

exports[`ConsoleLogOutputFormatter sectionSeparator 1`] = `
"
"
`;

exports[`ConsoleLogOutputFormatter snippet 1`] = `"\`code\`"`;
